journals/apps/journals/models.py

```diff
@@ -63,14 +80,92 @@ class Journal(models.Model):
     organization = models.ForeignKey(
         'Organization',
         on_delete=models.CASCADE,
-        null=True
+        null=False
     )
     access_length = models.DurationField()
+    objects = JournalManager()
+
+    class Meta(object):
+        unique_together = (
+            ('name', 'organization')
+        )
 
     def __str__(self):
         return self.name
 
 
+class JournalMetaData(object):
```

Comment:
I'm a little hesitant on the amount of duplicated data here, as it complicates what needs to be in sync. Other than price, currency, and sku, all of this data already lives in the Journal model. The marketing fields can be found via the JournalAboutPage attached to that Journal. I think it would be cleaner to hold the ecommerce info in the Journal model, and add the `_get_ecommerce_data()` and `_get_discovery_data()` methods there.